src/lib/components/image/Image.tsx

```diff
@@ -0,0 +1,171 @@
+import { useState } from "react";
+import classNames from "classnames";
+import { BiShow, BiError } from "react-icons/bi";
+import { VuiFlexContainer } from "../flex/FlexContainer";
+import { VuiFlexItem } from "../flex/FlexItem";
+import { VuiText } from "../typography/Text";
+import { VuiSkeleton } from "../skeleton/Skeleton";
+import { VuiImagePreview } from "./ImagePreview";
+import { VuiTooltip } from "../tooltip/Tooltip";
+import { VuiIcon } from "../icon/Icon";
+import { VuiTextColor } from "../typography/TextColor";
+
+export type ImageSize = "xs" | "s" | "m" | "l" | "xl" | "full";
+export type CaptionPosition = "top" | "bottom" | "overlay";
+
+type Props = {
+  src: string;
+  alt?: string;
+  caption?: string;
+  captionPosition?: CaptionPosition;
+  size?: ImageSize;
+  className?: string;
+  isLoading?: boolean;
+  allowPreview?: boolean;
+  errorMessage?: string;
+  previewTitle?: string;
+  previewDescription?: string;
+  previewCaption?: string;
+  controls?: {
+    allowRotation?: boolean;
+    allowZoom?: boolean;
+    allowReset?: boolean;
+    allowDrag?: boolean;
+  };
+};
+
+const sizeMap = {
+  xs: 100,
+  s: 200,
+  m: 400,
+  l: 600,
+  xl: 800,
+  full: "100%"
+} as const;
+
+const directionMap = {
+  top: "columnReverse",
+  bottom: "column"
+} as const;
+
+const getFlexDirection = (position: keyof typeof directionMap) => {
+  return directionMap[position];
+};
+
+const sizeConfig = {
```

Comment:
I think we should get rid of the `size` prop, and instead default the image to be 100% and auto height. This way the consumer can control the size through the container of the image. Adding support for a `className` prop that gets applied to the `img` element would be a nice escape hatch.

I'm not sure how that affects `skeletonRows`, maybe we just render a gray outline of the container for now if we can't render skeleton rows well?

I think we should also consistently use a `m` size.